File: challenges/magicalWell/python3/magicalWell.py
def magicalWell(a, b, n):
	s = 0
	while n > 0:
		n -= 1
		s += a*b
		a += 1
		b += 1
	return s

if __name__ == '__main__':
	input0 = [1, 1, 6, 1601, 1891, 1936, 957, 687, 491, 1275]
	input1 = [2, 1, 5, 337, 352, 1835, 57, 1337, 1552, 362]
	input2 = [2, 1, 3, 0, 0, 5, 2, 0, 4, 2]
	expectedOutput = [8, 1, 128, 0, 0, 17800540, 110113, 0, 3060400, 924738]
	assert len(input0) == len(expectedOutput), '# input0 = {}, # expectedOutput = {}'.format(len(input0), len(expectedOutput))
	assert len(input1) == len(expectedOutput), '# input1 = {}, # expectedOutput = {}'.format(len(input1), len(expectedOutput))
	assert len(input2) == len(expectedOutput), '# input2 = {}, # expectedOutput = {}'.format(len(input2), len(expectedOutput))
	for i, expected in enumerate(expectedOutput):
		actual = magicalWell(input0[i], input1[i], input2[i])
		assert actual == expected, 'magicalWell({}, {}, {}) returned {}, but expected {}'.format(input0[i], input1[i], input2[i], actual, expected)
	print('PASSES {} out of {} tests'.format(len(expectedOutput), len(expectedOutput)))

File: challenges/magicalWell/octave/magicalWell.m
function r = magicalWell(a, b, n)
	r = sum((a + [0:n-1]) .* (b + [0:n-1]));
endfunction
%!assert(magicalWell(1, 2, 2), 8)
%!assert(magicalWell(1, 1, 1), 1)
%!assert(magicalWell(6, 5, 3), 128)
%!assert(magicalWell(1601, 337, 0), 0)
%!assert(magicalWell(1891, 352, 0), 0)
%!assert(magicalWell(1936, 1835, 5), 17800540)
%!assert(magicalWell(957, 57, 2), 110113)
%!assert(magicalWell(687, 1337, 0), 0)
%!assert(magicalWell(491, 1552, 4), 3060400)
%!assert(magicalWell(1275, 362, 2), 924738)